feat: Add new assertions for URL paths

formula23 opened the pull request:
New operations for asserting paths:

```
$url = 'https://laravel.com/docs/11.x';

    $this->visit($url)
        ->assertPathBeginsWith('/docs');

    $this->visit($url)
        ->assertPathContains('/11');

    $this->visit($url)
        ->assertPathEndsWith('/11.x');

    $this->visit($url)
        ->assertPathIs('/docs/11.x');

//Assert path is with wildcard 

    $this->visit($url)
        ->assertPathIs('/docs/*.x');

    $this->visit($url)
        ->assertPathIsNot('/docs/12.x');
```

formula23 commented:
Once this PR is merged: [#63](https://github.com/pestphp/pest-plugin-browser/pull/63)

And everything else looks good, i can go in and change all the urls to use playground.

ijpatricio commented:
Hello @formula23 

Thank you for your work!

Can you please add Docs - check README.md for a guidance: https://github.com/pestphp/pest-plugin-browser?tab=readme-ov-file#available-assertions

And also you need to run `./vendor/bin/pint` to fix some unused imports.

Thanks!

formula23 commented:
Hi @ijpatricio, I have updated the README.md file with docs for theses assertions and addressed your other comments.

For some reason the test suite is failing on this test: 
` FAILED  Tests\Browser\Operations\ClickAtPointTest…  ProcessFailedException   `

formula23 commented:
Thanks @ijpatricio 
I’m wondering what’s the best way to continue contributing? Do you guys have any type of formal roadmap? Or should I just keep going down the list of assertions and adding ones that need completing? Also, are there any larger features that I can collaborate with you guys or assist with?

ijpatricio commented:
What a fantastic start @formula23 congrats 👏 

There's no formal roadmap yet.

The best direction for efforts is to figure out how to preserve the same Playwright instance/worker.

Why? For example, between tests, we need to reset a database, run a seeder, etc. If we were to start a new PW session (`npx playwright test`)  each time, the wasted time in these processes becomes huge.

There's an issue #12 with some starting directions.

I will comment there right now what I've been able to achieve - mostly I was managing PR's, but the higher goal is this right now. All assertions will have to be converted to the WS protocol, onde we figure how it works.

Although, yes, assertions and API are important, and if they're already in place with the JS compiler method, that's easier to convert later, and the test gets already there.

So, it's up to you on how you would prefer to contribute.

Thank you!

formula23 commented:
Thanks for the info @ijpatricio i'll look into the worker using a websocket and using its own process...

ijpatricio commented:
Thanks for the update @formula23  and for going on with helping on that task!

I'm also on and off tinkering with it, let's keep updated in #12  or  a new issue/PR if it makes sense!

formula23 commented:
@ijpatricio i pulled down the ws branch, i'll try to see whats going on there and move communication to #12
